types/pg/index.d.ts

```diff
@@ -9,10 +9,9 @@ import events = require("events");
 import stream = require("stream");
 import pgTypes = require("pg-types");
 
-// tslint:disable-next-line unified-signatures
-export function connect(connection: string, callback: (err: Error, client: Client, done: (err?: any) => void) => void): void;
-// tslint:disable-next-line unified-signatures
-export function connect(config: ClientConfig, callback: (err: Error, client: Client, done: (err?: any) => void) => void): void;
+export function connect(
+    connectionOrConfig: string | ClientConfig,
```

Comment:
I wish there was a name for this, for example `Configuration`. You could then do the following

```
type Configuration = string | Client config

export function connect(configuration: Configuration
```

You could also use it for the client constructor.

Reply:
Pg does not allow anymore to pass directly a connection string to the pool, but allows it as a property of the config object : 
```js
const pool = new Pool({
  connectionString:  'postgresql://dbuser:secretpassword@database.server.com:3211/mydb'
})
```
see https://node-postgres.com/features/connecting#connection-uri

Same thing for client config.
So we should add the config definition : 
```typescript
{
connectionString?: string
}
```

Reply:
@abenhamdine Got it. Please review the update.